### PR TITLE
#165305764 Favoriting an article

### PR DESCRIPTION
**Pull request by a-braham**
### What does this PR do?
- Allows users to favorite an article
### What are the tasks to be completed?
```
POST /api/articles/{slug}/favorite/
GET /api/articles/favorites/me/
DELETE /api/articles/{slug}/favorite/
```
* Ensure that a user can be able to post a favorite
* Ensure a user can be able to remove an article from favorites
* Ensure that user can be able to get his favorite articles
### How can this be manually tested?
 **After cloning the repo:-**
**Test with postman**
sign up a user, login grub the token and add it to authorization:
```
$ git checkout  ft-favorite-articles-165305764
$ python manage.py runserver
```
**Run tests on terminal**
```
$ python manage.py test
```

### What are the relevant pivotal tracker stories?
[#165305764](https://www.pivotaltracker.com/n/projects/2326460/stories/165305764)

### Screenshots
Screenshot 2019-05-08 at 12 28 14
Screenshot 2019-05-09 at 14 30 48
Screenshot 2019-05-09 at 14 31 00

### CODE REVIEW

**Comment by Bakley** [comment]
@a-braham  Kindly add **How to run test** on your PR descriptions. It will help new users know what to do to see the test passing locally

**Review by matthenge — CHANGES_REQUESTED** [review]
@a-braham  Please update the PR comment to reflect the delete endpoint. Also, change the screenshot for the delete endpoint to reflect the current one

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
+from django.shortcuts import render
```
Comment (Ogutu-Brian) [inline authors/apps/articles/views.py]: Hello @a-braham ,
You are not using this module, so you don't need to import it

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]

File: authors/apps/articles/tests/test_favorite.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    def test_get_favorite_successfully(self):
+        """
+        Test method for successfully deleting favorites
+        """
+        self.is_authenticated("adam@gmail.com", "@Us3r.com")
+        self.post_favorite()
+        response = self.get_favorites()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data.get("favorites")[0].get("slug"), self.article.slug)    
+
```
Comment (Ogutu-Brian) [inline authors/apps/articles/tests/test_favorite.py:77]: Hello @a-braham ,
Please add a new line at the end of file

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]

**Review by matthenge — APPROVED** [review]
LGTM

**Review by IssaIan — APPROVED** [review]
LGTM

**Pull request merged**